app/infrastructure/messaging/websocket_manager.py

```
@@ -1,46 +1,350 @@
+    
+    async def send_to_vault(self, vault_id: str, message: dict):
+        """Send message to a specific vault device"""
+        if vault_id not in self.vault_connections:
+            logger.warning(f"Vault {vault_id} is not connected")
+            raise ConnectionError(f"Vault {vault_id} is offline")
+        
+        try:
+            await self.vault_connections[vault_id].send_json(message)
+            logger.debug(f"Sent message to vault {vault_id}: {message['type']}")
+        except Exception as e:
+            logger.error(f"Failed to send to vault {vault_id}: {e}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)